processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
@leventov you don't have to create column >2GB to perform v2 benchmark, you can adjust https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java#L77 locally to create v2 GenericIndexed. Try to adjust it to something which creates 2-3 value files.